tough/src/error.rs

```diff
@@ -176,12 +230,42 @@ pub enum Error {
         backtrace: Backtrace,
     },
 
+    /// Path isn't a valid UTF8 string
+    #[snafu(display("Path {} is not valid UTF-8", path.display()))]
+    PathUtf8 { path: PathBuf, backtrace: Backtrace },
+
+    #[snafu(display("Failed to serialize role for signing: {}", source))]
+    SerializeRole {
+        source: serde_json::Error,
+        backtrace: Backtrace,
+    },
+
+    #[snafu(display("Failed to serialize signed role: {}", source))]
+    SerializeSignedRole {
+        source: serde_json::Error,
+        backtrace: Backtrace,
+    },
+
     #[snafu(display("Failed to sign message"))]
     Sign {
         source: ring::error::Unspecified,
         backtrace: Backtrace,
     },
 
+    #[snafu(display("Unable to find signing keys for role '{}'", role))]
+    SigningKeysNotFound { role: String },
+
+    #[snafu(display(
+        "Tried to load repository with spec version '{}', version '{}' is supported",
```

Comment:
Not necessarily a repository; we also use this error when we try to set targets, which you can do in a fresh editor.